### PR TITLE
WIP: Correlations: Enable in Dashboards

### PR DESCRIPTION
**Pull request by gelicia**
This is a draft while I clean things up and test more panel types

**What is this feature?**

[Add a brief description of what the feature or update does.]

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: packages/grafana-data/src/field/fieldOverrides.ts [hunk 1]
```diff
@@ -436,33 +469,90 @@ export const getLinksSupplier =
...
+
+        // add all fields in frame as variables
+        frame.fields.forEach((f) => {
+          if (fieldDisplayValuesProxy && fieldDisplayValuesProxy[f.name]) {
+            rowVars[f.name] = {
+              value: fieldDisplayValuesProxy[f.name],
+            };
+          }
+        });
+      }
```
Comment (torkelo) [inline packages/grafana-data/src/field/fieldOverrides.ts:498]: Need to figure out how to do this through a variable macro instead (they are lazily evaluated), building every possible var by building a large scopedVar object every time is now how we do it anymore (data macros replaced all of this). https://github.com/grafana/scenes/blob/main/packages/scenes/src/variables/macros/dataMacros.ts

File: packages/grafana-data/src/field/fieldOverrides.ts [hunk 2]
```diff
@@ -436,33 +469,90 @@ export const getLinksSupplier =
...
+          if (config.valueRowIndex !== undefined) {
+            if (transformation.field) {
+              const transformField = frame?.fields.find((field) => field.name === transformation.field);
+              fieldValue = transformField?.values[config.valueRowIndex];
+            } else {
+              fieldValue = field.values[config.valueRowIndex];
+            }
+          } else if (config.calculatedValue) {
+            fieldValue = config.calculatedValue;
+          }
```
Comment (torkelo) [inline packages/grafana-data/src/field/fieldOverrides.ts:514]: Unsure what is going on here and how this is needed. 

 what's the problem/scenario (unit tests for applyFieldOverrides or getLinkSupplier would help.

File: packages/grafana-data/src/field/fieldOverrides.ts [hunk 3]
```diff
@@ -436,33 +469,90 @@ export const getLinksSupplier =
...
+          const replaceInfo = enhancedReplaceVariables!(value, { ...dataLinkScopedVars, ...scopedVars }, format);
+          if (replaceInfo.allFound) {
+            return replaceInfo.replaceStr;
+          } else {
+            return undefined;
+          }
+        } else {
+          return replaceVariables(value, { ...dataLinkScopedVars, ...scopedVars }, format);
+        }
+      };
```
Comment (torkelo) [inline packages/grafana-data/src/field/fieldOverrides.ts:544]: What is this enhanced thing?

**Review by torkelo — COMMENTED** [review]
WIsh you had reached out to dashboard squad with some guidance first, the old dashboard arch is in code freeze. So all the changes to the old annotations/dashboard query workers are not needed (as they are for the old architecture).

A bit worried about the performance of doing this to every dataframe on every process step (So trying to do as much as possible lazily is important).  

Wonder if we can use correlations to do more than just link to explore, would want it to be able to open views / change layout in dashboard as well (show a drilldiown panel etc)

**Comment by gelicia** [comment]
> Wonder if we can use correlations to do more than just link to explore, would want it to be able to open views / change layout in dashboard as well (show a drilldiown panel etc)


Basic functionality to link to a url is available now. Building this out to do dashboard specific tasks is something we can look into if we want to invest more in it https://github.com/grafana/grafana/pull/92442

**Pull request closed without merging**